pgxn/neon/pagestore_smgr.c

```diff
@@ -1604,19 +1604,23 @@ neon_write(SMgrRelation reln, ForkNumber forknum, BlockNumber blocknum, const vo
 #endif
 			{
 				/* It exists locally. Guess it's unlogged then. */
+				if (mdnblocks(reln, forknum) >= blocknum)
+				{
+					/* prevent race condition with unlogged build end which unlinks local files */
```

Comment:
This needs a comment to explain why `mdnblocks` is called. The caller is never supposed to call smgrwrite() with a block that's beyond the EOF, so this call looks pointless.

Reply:
Done